app/views/play/level/tome/SpellView.coffee

```diff
@@ -711,7 +712,8 @@ module.exports = class SpellView extends CocoView
 
   recompile: (cast=true, realTime=false, cinematic=false) ->
     @fetchTokenForSource().then (source) =>
-      hasChanged = @spell.source isnt source
+      readableSource = Aether.getTokenSource(source)
```

Comment:
Can we call `getTokenSource` only if `language` is cpp (or not python/js) to avoid regression risk?
Similarly for lines 102-103 above in aether.coffee

Reply:
This should not lead to regressions as the algorithm passes through the raw text provided it doesn't trigger the unicode check.